pipelines/test/headers.yaml

```diff
@@ -0,0 +1,76 @@
+name: headers
+
+needs:
+  packages:
+    - autoconf
+    - build-base
+    - file
+
+pipeline:
+  - name: syntax check .h headers
+    runs: |
+      function test_header() {
+        local lang="$1"
+        local header="$2"
+        local rc=1
+        d=$(mktemp -d XXXXXXXX)
+        cd $d
+        case $lang in
+          c)
+            gen_c_ac "$header"
+          ;;
+          cpp)
+            gen_cpp_ac "$header"
+          ;;
+          *)
+            return 1
+          ;;
+        esac
+        cat configure.ac
+        autoconf
+        ./configure && rc=0 || rc=1
+        cat config.log
+        cd ..
+        rm -rf $d
+        return $rc
+      }
+      function gen_c_ac() {
+        local f="$1"
+        cat >configure.ac <<EOF
+        AC_INIT([example],[1.0],[example@example.com])
+        AC_PREREQ(2.69)
+        AC_PROG_CC
+        AC_LANG(C)
+        AC_CHECK_HEADER($f,[c_header_found=yes],,)
+        AS_IF([test "x\$c_header_found" != "xyes"], [AC_MSG_ERROR([Unable to check header])])
+        AC_OUTPUT()
+      EOF
+      }
+      function gen_cpp_ac() {
+        local f="$1"
+        cat >configure.ac <<EOF
+        AC_INIT([example],[1.0],[example@example.com])
+        AC_PREREQ(2.69)
+        AC_PROG_CXX
+        AC_LANG(C++)
+        AC_CHECK_HEADER($f,[cxx_header_found=yes],,)
+        AS_IF([test "x\$cxx_header_found" != "xyes"], [AC_MSG_ERROR([Unable to check header])])
+        AC_OUTPUT()
+      EOF
+      }
+      pkg=$(basename ${{targets.contextdir}})
```

Comment:
You can use context.name now.

Reply:
Thanks, will do.  I'm moving the vast majority of this logic to tw.

Reply:
Moved all of this to tw.